File: backend/api/cron/v1/cron.go
package v1

import (
	"backend/internal/model/entity"

	"github.com/gogf/gf/v2/frame/g"
)

type CronCreateReq struct {
	g.Meta            `path:"/v1/cronCreate" method:"post"`
	CronName          string `json:"cron_name" v:"required"`
	KnowledgeBaseName string `json:"knowledge_base_name" v:"required"`
	CronExpression    string `json:"cron_expression" v:"required"`
	SchedulingMethod  string `json:"scheduling_method" v:"required"`
	Status            int64  `json:"status" v:"required"`
	ContentType       int64  `json:"content_type" v:"required"`
}
type CronCreateRes struct {
	ID string `json:"id"` // UUID
}

type CronDeleteReq struct {
	g.Meta `path:"/v1/cronDelete" method:"delete"`
	ID     string `json:"id" v:"required"` // UUID
}
type CronDeleteRes struct {
	IsOK string `json:"is_ok"`
}

type CronListReq struct {
	g.Meta `path:"/v1/cronList" method:"get"`
	Page   int `p:"page" dc:"page" v:"required|min:1" d:"1"`
	Size   int `p:"size" dc:"size" v:"required|min:1|max:100" d:"10"`
}
type CronListRes struct {
	List []entity.KnowledgeBaseCronSchedule `json:"list"`
}

type CronGetOneReq struct {
	g.Meta `path:"/v1/cronGetOne" method:"get"`
	Id     string `json:"id"` // UUID
}
type CronGetOneRes struct {
	One *entity.KnowledgeBaseCronSchedule `json:"one"`
}

type CronUpdateOneReq struct {
	g.Meta            `path:"/v1/cronUpdateOne" method:"put"`
	Id                string `json:"id"` // UUID
	CronName          string `json:"cron_name" v:"required"`
	KnowledgeBaseName string `json:"knowledge_base_name" v:"required"`
	CronExpression    string `json:"cron_expression" v:"required"`
	Status            int64  `json:"status" v:"required"`
	ContentType       int64  `json:"content_type" v:"required"`
	SchedulingMethod  string `json:"scheduling_method" v:"required"`
}
type CronUpdateOneRes struct {
	IsOK string `json:"is_ok"`
}

type CronUpdateStatusReq struct {
	g.Meta `path:"/v1/cronUpdateOneStatus" method:"put"`
	Id     string `json:"id"` // UUID
	Status int64  `json:"status" v:"required"`
}
type CronUpdateStatusRes struct {
	IsOK string `json:"is_ok"`
}

type CronRunReq struct {
	g.Meta `path:"/v1/cronRun" method:"post"`
	Id     string `json:"id" v:"required"` // UUID
}

type CronRunRes struct {
	IsOK string `json:"is_ok"`
}

File: frontChat/src/services/cron.ts
import http from '@/utils/axios';

export interface CronCreateReq {
    cron_name: string;
    knowledge_base_name: string;
    scheduling_method: string;
    cron_expression: string;
    status: number;
    content_type: number;
}

export interface CronCreateRes {
    id: number;
}

export interface CronDeleteReq {
    id: number;
}

export interface CronDeleteRes {
    is_ok: string;
}

export interface CronListReq {
    page: number;
    size: number;
}

export interface KnowledgeBaseCronSchedule {
    id: number;
    cron_name: string;
    cronName?: string; // Compatible with camelCase
    knowledge_base_name: string;
    knowledgeBaseName?: string; // Compatible with old backend response
    scheduling_method: string;
    schedulingMethod?: string; // Compatible with camelCase
    cron_expression: string;
    cronExpression?: string; // Compatible with camelCase
    status: number;
    content_type: number;
    contentType?: number; // Compatible with camelCase
    created_at: string;
    createdAt?: string; // Compatible with camelCase
    updated_at: string;
    updatedAt?: string; // Compatible with camelCase
    deleted_at?: string;
    deletedAt?: string; // Compatible with camelCase
}

export interface CronListRes {
    list: KnowledgeBaseCronSchedule[];
}

export interface CronGetOneReq {
    id: number;
}

export interface CronGetOneRes {
    one: KnowledgeBaseCronSchedule;
}

export interface CronUpdateOneReq {
    id: number;
    cron_name?: string;
    knowledge_base_name?: string;
    scheduling_method?: string;
    cron_expression?: string;
    status?: number;
    content_type?: number;
}

export interface CronUpdateOneRes {
    is_ok: string;
}

export interface CronUpdateOneStatusReq {
    id: number;
    status: number;
}

export interface CronUpdateOneStatusRes {
    is_ok: string;
}

export interface CronRunReq {
    id: number;
}

export interface CronRunRes {
    is_ok: string;
}

export interface CronExecuteListReq {
    cron_name_fk: string;
    page: number;
    size: number;
}

export interface CronExecute {
    id: number;
    cron_id: number;
    cronId?: number; // Compatible with camelCase
    cron_name_fk: string;
    cronNameFk?: string; // Compatible with camelCase
    execute_time: string;
    executeTime?: string; // Compatible with camelCase
    next_time?: string;
    nextTime?: string; // Compatible with camelCase
    status: number; // 执行状态：0=执行中，1=成功，2=失败
    error_message?: string;
    errorMessage?: string; // Compatible with camelCase
    duration?: number; // 执行耗时（毫秒）
    created_at: string;
    createdAt?: string; // Compatible with camelCase
    updated_at: string;
    updatedAt?: string; // Compatible with camelCase
}

export interface CronLog {
    id: number;
    execute_id: number;
    executeId?: number; // Compatible with camelCase
    cron_id: number;
    cronId?: number; // Compatible with camelCase
    cron_name_fk: string;
    cronNameFk?: string; // Compatible with camelCase
    content: string;
    level: string;
    create_time: string;
    createTime?: string; // Compatible with camelCase
}

export interface CronExecuteListByCronIdReq {
    cronId: number;
    page: number;
    size: number;
}

export interface CronExecuteListByCronIdRes {
    list: CronExecute[];
    total: number;
}

export interface CronExecuteDetailReq {
    id: number;
}

export interface CronExecuteDetailRes extends CronExecute {}

export interface CronExecuteLogReq {
    executeId: number;
    page: number;
    size: number;
}

export interface CronExecuteLogRes {
    list: CronLog[];
    total: number;
}

export interface CronExecuteListRes {
    list: CronExecute[];
    total: number;
}

export const CronService = {
    create: (data: CronCreateReq) => {
        return http.post<CronCreateRes>('/gateway/v1/cronCreate', data);
    },
    delete: (data: CronDeleteReq) => {
        return http.delete<CronDeleteRes>('/gateway/v1/cronDelete', { data });
    },
    list: (params: CronListReq) => {
        return http.get<CronListRes>('/gateway/v1/cronList', params);
    },
    getOne: (params: CronGetOneReq) => {
        return http.get<CronGetOneRes>('/gateway/v1/cronGetOne', params);
    },
    updateOne: (data: CronUpdateOneReq) => {
        return http.put<CronUpdateOneRes>('/gateway/v1/cronUpdateOne', data);
    },
    updateOneStatus: (data: CronUpdateOneStatusReq) => {
        return http.put<CronUpdateOneStatusRes>('/gateway/v1/cronUpdateOneStatus', data);
    },
    run: (data: CronRunReq) => {
        return http.post<CronRunRes>('/gateway/v1/cronRun', data);
    },
    listLogs: (params: CronExecuteListReq) => {
        return http.get<CronExecuteListRes>('/gateway/v1/cronExecuteList', params);
    },
    listByCronId: (params: CronExecuteListByCronIdReq) => {
        return http.get<CronExecuteListByCronIdRes>('/gateway/v1/cronExecuteListByCronId', params);
    },
    getExecuteDetail: (params: CronExecuteDetailReq) => {
        return http.get<CronExecuteDetailRes>('/gateway/v1/cronExecuteDetail', params);
    },
    getExecuteLogs: (params: CronExecuteLogReq) => {
        return http.get<CronExecuteLogRes>('/gateway/v1/cronExecuteLog', params);
    }
};
